msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -1120,8 +1147,43 @@ internal void ClearAdalCache()
         }
 
         internal void ClearMsalCache()
+        {
+            try
+            {
+                TokenCacheNotificationArgs args = new TokenCacheNotificationArgs
+                {
+                    TokenCache = this,
+                    ClientId = ClientId,
+                    Account = null
+                };
+
+                OnBeforeAccess(args);
```

Comment:
>OnBeforeAccess [](start = 16, length = 14)

Why do you need this? Do you call ClearMsalCache on its own? I believe you should call Clear() that will call both ClearMsalCache and ClearAdalCache (this is to ensure that the unified cache is consistent). I agree that ClearMsalCache and ClearAdalCache should have been made private and not internal, as they are not meant to called from outside.

Reply:
From my point of view, this is the only outstanding issue. 

---
In reply to: [229642843](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1293#discussion_r229642843) [](ancestors = 229642843)

Reply:
This is not my change...might be something from a merge or accidental formatting change